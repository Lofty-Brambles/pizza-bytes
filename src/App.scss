@use "styles/mix";
@use "styles/var";

.App {}