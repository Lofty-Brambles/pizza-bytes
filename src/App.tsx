import React from "react";
import "./App.scss";
import { Home, Menu, ShoppingBag } from "react-feather";
import { Outlet } from "react-router-dom";
import page from "./interfaces";

import Homepg from "./pages/Homepg";
import FootBar from "./components/FootBar";
import TitleBar from "./components/TitleBar";

import Routing from "./Routing";
import Menupg from "./pages/Menupg";
import Cartpg from "./pages/Cartpg";

const dtls: Array<page> = [
	{
		name: "Home",
		path: "/",
		component: <Homepg />,
		icon: <Home color="red" size={24} />,
	},
	{
		name: "Menu",
		path: "/menu",
		component: <Menupg />,
		icon: <Menu color="blue" size={24} />,
	},
	{
		name: "Cart",
		path: "/cart",
		component: <Cartpg />,
		icon: <ShoppingBag color="purple" size={24} />,
	},
];

const App = () => (
	<div className="App">
		<TitleBar />
		<Routing pages={dtls} />
		<main>
			<Outlet />
		</main>
		<FootBar pages={dtls} />
	</div>
);

export default App;
export { dtls };
