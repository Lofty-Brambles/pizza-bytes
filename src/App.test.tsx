/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("Main App Component", () => {
	test("rendering", () => {
		const { container } = render(<App />);
		expect(container).toMatchSnapshot();
	});
});
