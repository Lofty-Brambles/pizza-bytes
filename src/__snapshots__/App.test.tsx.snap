// Vitest Snapshot v1

exports[`Main App Component > rendering 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <img
        alt="logo"
        src="./pizza-logo.png"
      />
      <h1>
        Pizza-Bytes
      </h1>
    </header>
  </div>
</div>
`;
